artemis-core-client/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/ChannelImpl.java

```diff
@@ -451,7 +451,9 @@ public Packet sendBlocking(final Packet packet,
             }
 
             if (response == null) {
-               throw ActiveMQClientMessageBundle.BUNDLE.timedOutSendingPacket(connection.getBlockingCallTimeout(), packet.getType());
+               ActiveMQException e = ActiveMQClientMessageBundle.BUNDLE.timedOutSendingPacket(connection.getBlockingCallTimeout(), packet.getType());
+               connection.fail(e);
```

Comment:
LGTM: I would just run the whole testsuite to make sure. I can run it here and merge it ok.

ok with you?

Reply:
I'm more concerned on FailoverTests... if the connection.failure here would cause any further issues.

Reply:
So i ran a few of them in the normal int tests (didnt run any extra tests tho). Happy for you to run more if you like

Reply:
@michaelandrepearce I ran the 2 hours:40 min test with on a jenkins server I have:
```
mvn test -Ptests
```

apparently there are tests hunging because of this.


I couldn't figure out what test is it.. as the build generate zillions of:

```
med out after waiting 2,000 ms for response when sending packet 32 [code=CONNECTION_TIMEDOUT]
[main] 09:19:18,204 WARN  [org.apache.activemq.artemis.core.client] AMQ212037: Connection failure has been detected: AMQ219014: Timed out after waiting 2,000 ms for response when sending packet 32 [code=CONNECTION_TIMEDOUT]
2019-05-01 09:19:18,204 WARN  [org.apache.activemq.artemis.core.client] AMQ212037: Connection failure has been detected: AMQ219014: Timed out after waiting 2,000 ms for response when sending packet 32 [code=CONNECTION_TIMEDOUT]
2019-05-01 09:19:18,204 WARN  [org.apache.activemq.artemis.core.client] AMQ212037: Connection failure has been detected: AMQ219014: Timed out after waiting 2,000 ms for response when sending packet 32 [code=CONNECTION_TIMEDOUT]
[main] 09:19:20,210 WARN  [org.apache.activemq.artemis.core.client] AMQ212037: Connection failure has been detected: AMQ219014: Timed out after waiting 2,000 ms for response when sending packet 32 [code=CONNECTION_TIMEDOUT]
2019-05-01 09:19:20,210 WARN  [org.apache.activemq.artemis.core.client] AMQ212037: Connection failure has been detected: AMQ219014: Timed out after waiting 2,000 ms for response when sending packet 32 [code=CONNECTION_TIMEDOUT]
2019-05-01 09:19:20,210 WARN  [org.apache.activemq.artemis.core.client] AMQ212037: Connection failure has been detected: AMQ219014: Timed out after waiting 2,000 ms for response when sending packet 32 [code=CONNECTION_TIMEDOUT]
[main] 09:19:22,216 WARN  [org.apache.activemq.artemis.core.client] AMQ212037: Connection failure has been detected: AMQ219014: Timed out after waiting 2,000 ms for response when sending packet 32 [code=CONNECTION_TIMEDOUT]
2019-05-01 09:19:22,216 WARN  [org.apache.activemq.artemis.core.client] AMQ212037: Connection failure has been detected: AMQ219014: 
```


I am taking a break today.. I can take a look on more details later today / tomorrow morning.

Reply:
I bet it will be nettyFailoverTest. I have a Deja Voux on this. I believe I had tried once to add such code and the reconnection loop failed.

Reply:
It is not NettyFailoverTest. ran it manually.

I will have to read the logs later today / tomorrow morning.

Keep this PR open please until we figure out?

Reply:
@michaelandrepearce org.apache.activemq.artemis.tests.integration.remoting.ReconnectTestis::testReattach failing.

It's probably a test issue, since the test was meant to disconnect the inner socket and play with a reattachment.

Reply:
@clebertsuconic ill keep pr open, if we can't resolve obviously we cannot merge, but then possibly will need to find another way to fix this.

Reply:
@clebertsuconic have just pushed a small change to that test, originally it would just repeat the same over and over, where as test is just expecting it ever to be called once, so made the interceptor in the test to adjust the behaviour just the once using a bool

Reply:
@clebertsuconic if you look at and let me know what you think.